### PR TITLE
HTCONDOR-2449 Add -long, -format and -af to condor_ping

### PR DESCRIPTION
**Pull request by johnkn**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/version-history/feature-versions-23-x.rst [hunk 1]
```diff
@@ -17,10 +17,15 @@ Release Notes:
 
 New Features:
 
-- Added an ``-edit`` option to the *condor_qusers* tool.  This option allows
+- Added an ``-edit`` option to the :tool:`condor_qusers`.  This option allows
   and administrator to add custom attributes to a User classad in the *condor_schedd*.
   :jira:`2381`
 
+- Added an ``-long``, ``-format`` and ``-autoformat`` options to the :tool:`condor_ping`.
+  These options allows give predictable output for programs that wish to parse the results
```
Comment (ColeBollig) [inline docs/version-history/feature-versions-23-x.rst:25]:
```suggestion
  These options give predictable output for programs that wish to parse the results
```

File: src/condor_tools/ping.cpp [hunk 2]
```diff
@@ -777,6 +925,55 @@ int main( int argc, const char *argv[] )
...
+				usage(App.Name);
+				exit(1);
+			}
+			App.prmask.registerFormatF(argv[i+1], argv[i+2], FormatOptionNoTruncate);
+			if ( ! IsValidClassAdExpression(argv[i+2], &App.prattrs, NULL)) {
+				fprintf(stderr,"ERROR: %s is not a valid expression\n\n", argv[i+2]);
+				usage(App.Name);
+				exit(1);
+			}
+			output_mode = 3;
```
Comment (ColeBollig) [inline src/condor_tools/ping.cpp:956]: Above conditional checks for `output_mode == 4` but is being set to `3`. Is the `-format` flag supposed to set output_made to `3` (long) or `4` (af)?

Comment (johnkn) [reply src/condor_tools/ping.cpp:956]: line should be removed.

**Review by ColeBollig — APPROVED** [review]
LGTM... minor changes should not impact Batlab and previous run was successful

**Pull request merged**